Xamarin.Forms.Xaml/MarkupExtensions/OnIdiomExtension.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+
+namespace Xamarin.Forms.Xaml
+{
+	public class OnIdiomExtension : IMarkupExtension
+	{
+		// See Device.Idiom
+
+		public object Default { get; set; }
+		public object Unsupported { get; set; }
+		public object Phone { get; set; }
+		public object Tablet { get; set; } 
+		public object Desktop { get; set; }
+		public object TV { get; set; }
+		public object Watch { get; set; }
+
+		public object ProvideValue(IServiceProvider serviceProvider)
```

Comment:
the Xaml engine expects from `ProvideValue` a value of the correct type that can be directly assigned to the property. In this case, all the objects are of type string, and not directly assignable to `FontSize`, `TextColor`, `WidthRequest`...

the magic of the runtime Xaml loader compensates for the wrong type and do the type conversion, but
 - I can't guarantee that to work in all cases
 - if fails with XamlC on

Reply:
Addressed in latest commits. Thanks!